pylot/planning/frenet_optimal_trajectory/FrenetOptimalTrajectory/src/FrenetOptimalTrajectory/fot_wrapper.cpp

```diff
@@ -0,0 +1,113 @@
+#include <vector>
+#include <iostream>
+#include "../utils.h"
+#include "../CubicSpline/CubicSpline2D.h"
+#include "FrenetOptimalTrajectory.h"
+
+using namespace std;
+
+// C++ wrapper to expose the FrenetOptimalTrajectory class to python
+extern "C" {
+    // Compute the frenet optimal trajectory given initial conditions
+    // in frenet space.
+    //
+    // Arguments:
+    //      s0: initial longitudinal position along spline
+    //      c_speed: initial velocity
+    //      c_d: initial lateral offset
+    //      c_d_d: initial lateral velocity
+    //      c_d_dd: initial lateral acceleration
+    //      xp, yp: list of waypoint coordinates
+    //      xo, yo: list of obstacle coordinates
+    //      np, no: length of the waypoint list and obstacle list
+    //      target_speed: target velocity in [m/s]
+    // Returns:
+    //      x_path, y_path: the frenet optimal trajectory in cartesian space
+    //      misc: the next states s0, c_speed, c_d, c_d_d, c_d_dd
+    int get_fot_frenet_space(
+            double s0, double c_speed, double c_d, double c_d_d, double c_d_dd,
+            double* xp, double* yp, double* xo, double* yo, int np, int no,
+            double target_speed, double* x_path, double* y_path,
+            double* speeds, double* misc
+            ) {
+        vector<double> wx (xp, xp + np);
+        vector<double> wy (yp, yp + np);
+
+        vector<tuple<double, double>> obstacles;
+        vector<double> ox (xo, xo + no);
+        vector<double> oy (yo, yo + no);
+        for (int i = 0; i < ox.size(); i++) {
+            tuple<double, double> ob (ox[i], oy[i]);
+            obstacles.push_back(ob);
+        }
+
+        FrenetOptimalTrajectory fot = FrenetOptimalTrajectory(wx, wy, s0,
```

Comment:
I find a bit strange that this is a class given that you never reuse it. It looks like a function to me :).

Reply:
True, I think I just found it easier / cleaner to implement because there's lots of attributes. Otherwise I'd have to pass them around different function calls or write a very large function right?